swupd/delta.go

```diff
@@ -31,149 +54,122 @@ func CreateDeltas(manifest, statedir string, from, to uint32) ([]*File, error) {
 		return nil, err
 	}
 
-	// Must be sorted before passing into linkPeersAndChange()
-	oldManifest.sortFilesName()
-	newManifest.sortFilesName()
-
-	_, _, _ = newManifest.linkPeersAndChange(oldManifest, from)
+	return createDeltasFromManifests(&c, oldManifest, newManifest)
+}
 
-	deltas, err := consolidateDeltaFiles(newManifest, from, c)
+func createDeltasFromManifests(c *config, oldManifest, newManifest *Manifest) ([]Delta, error) {
+	deltas, err := findDeltas(c, oldManifest, newManifest)
 	if err != nil {
-		return nil, errors.Wrapf(err, "Failed to create deltas list %s", manifest)
+		return nil, errors.Wrapf(err, "Failed to create deltas list %s", newManifest.Name)
+	}
+
+	if len(deltas) == 0 {
+		return []Delta{}, nil
 	}
 
+	// TODO: Unify these maxRoutines constants, get a better default (maybe NumCPUs)
```

Comment:
In the C code we have typically used 3 or NumCPUs*3, and my (very minimal) testing showed that it was a consistently good limit. I'd be interested in your perf analysis later on to see if this still holds true here.

Reply:
Just making the user has slightly more control of this (even if our default remains 3) will already be useful.